clustered/server/src/main/java/org/ehcache/clustered/server/state/EhcacheStateContext.java

```diff
@@ -0,0 +1,26 @@
+/*
+ * Copyright Terracotta, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.ehcache.clustered.server.state;
+
+import com.tc.classloader.CommonComponent;
+
+/**
+ * Marker interface to pass context between begin and end message processing.
+ */
+@CommonComponent
+public interface EhcacheStateContext extends AutoCloseable {
```

Comment:
This seems awfully heavyweight just to avoid catching an Exception in the outer log... can't we just use AutoCloseable here and amend the exception handling logic in the call sites?

Reply:
@chrisdennis , this simplifies exception handling in ClusterTierPassiveEntity nicely as otherwise I need to differentiate between cluster exception and all other exceptions. Thought it was more readable/extensible as well with future possibilities to extend the interface to add additional behaviour to the context (e.g notifying stages of processing, if any).